File: java/src/test/java/model/CoordinatePairTest.java
package model;

import static org.junit.jupiter.api.Assertions.assertEquals;

import org.junit.jupiter.api.Test;

class CoordinatePairTest {

    @Test
    void shouldBuildCoordinatePairObjectFromIntegerCoordinates() {
        // When
        final CoordinatePair coordinatePair = CoordinatePair.of(2, 3);

        // Then
        assertEquals(ImmutableCoordinatePair.class, coordinatePair.getClass());
        assertEquals(2, coordinatePair.getXCoordinate());
        assertEquals(3, coordinatePair.getYCoordinate());
    }
}

File: scala/src/test/scala/model/CoordinatePairTest.scala
package model

import org.scalatest.funsuite.AnyFunSuite

class CoordinatePairTest extends AnyFunSuite {

  test("should build CoordinatePair object from integer coordinates") {
    // When
    val coordinatePair: CoordinatePair = CoordinatePair(2, 3)

    // Then
    assert(coordinatePair.isInstanceOf[CoordinatePair])
    assert(2 == coordinatePair.xCoordinate)
    assert(3 == coordinatePair.yCoordinate)
  }
}
